llvm/lib/CodeGen/BasicBlockSections.cpp

```diff
@@ -99,6 +101,33 @@ static cl::opt<bool> BBSectionsDetectSourceDrift(
 
 namespace {
 
+MachineBasicBlock *CloneMachineBasicBlock(MachineBasicBlock *MBB) {
```

Comment:
Why can't this be part of the IR itself, why not in BasicBlock.h?

Reply:
We need to copy the machine-instructions.

Reply:
I mean CodeGen/MachineBasicBlock.h.   This is a simple change so I am not insisting.

Reply:
I see. MachineFunction.h is the proper place. I augmented `MachineFunction::createMachineBasicBlock` so we can piggy-back on it.